x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ShrinkStep.java

```diff
@@ -21,19 +22,24 @@
 public class ShrinkStep extends AsyncActionStep {
     public static final String NAME = "shrink";
 
-    private int numberOfShards;
+    private Integer numberOfShards;
+    private ByteSizeValue maxSinglePrimaryShardSize;
     private String shrunkIndexPrefix;
 
-    public ShrinkStep(StepKey key, StepKey nextStepKey, Client client, int numberOfShards, String shrunkIndexPrefix) {
+    public ShrinkStep(StepKey key, StepKey nextStepKey, Client client, Integer numberOfShards,
+                      ByteSizeValue maxSinglePrimaryShardSize, String shrunkIndexPrefix) {
         super(key, nextStepKey, client);
         this.numberOfShards = numberOfShards;
+        this.maxSinglePrimaryShardSize = maxSinglePrimaryShardSize;
         this.shrunkIndexPrefix = shrunkIndexPrefix;
     }
 
-    public int getNumberOfShards() {
+    public Integer getNumberOfShards() {
         return numberOfShards;
     }
 
+    public ByteSizeValue getMaxSinglePrimaryShardSize() {return maxSinglePrimaryShardSize;}
```

Comment:
Super minor, but we usually don't collapse these quite so much, so I think

```suggestion
    public ByteSizeValue getMaxSinglePrimaryShardSize() {
        return maxSinglePrimaryShardSize;
    }
```

Is probably best for now